docs/aws/audit/esmonitoring/rules/es_encrypted_with_cmks_remediation.mdx

```diff
@@ -1,6 +1,198 @@
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon ES
+   client = boto3.client('es')
+
+   # Define the domain name and KMS Key ID
+   domain_name = 'your-elasticsearch-domain'
+   kms_key_id = 'your-kms-key-id'
+
+   # Create or update the Elasticsearch domain with encryption enabled
+   response = client.create_elasticsearch_domain(
+       DomainName=domain_name,
+       EncryptionAtRestOptions={
+           'Enabled': True,
+           'KmsKeyId': kms_key_id
+       }
+   )
+
+   print(response)
+   ```
+
+4. **Run the Script**:
+   Execute the script to ensure that your Elasticsearch domain is created or updated with encryption enabled using the specified KMS CMK.
+
+   ```bash
+   python your_script_name.py
+   ```
+
+By following these steps, you can ensure that your Elasticsearch domain is encrypted with KMS CMKs, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon Elasticsearch Service dashboard. 
+3. In the navigation pane, choose "Domains".
+4. For each domain listed in the Domains table, check the Encryption column. If the column displays "Disabled", the selected Amazon Elasticsearch domain is not encrypted with KMS CMKs.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the Elasticsearch domains.
+
+2. Use the following AWS CLI command to list all the Elasticsearch domains:
+
+   ```
+   aws es list-domain-names
+   ```
+   This command will return a list of all Elasticsearch domain names.
+
+3. For each domain, use the following command to describe the Elasticsearch domain configuration:
+
+   ```
+   aws es describe-elasticsearch-domain --domain-name <your-domain-name>
+   ```
+   Replace `<your-domain-name>` with the name of your Elasticsearch domain. This command will return the configuration details of the specified Elasticsearch domain.
+
+4. Check the output of the above command for the `EncryptionAtRestOptions` field. If the `Enabled` field under `EncryptionAtRestOptions` is `false`, or if the `KmsKeyId` field is not present or empty, then the Elasticsearch domain is not encrypted with KMS CMKs.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+
+2. Create a session using your AWS credentials. You can configure your credentials in the AWS credentials file. 
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>